sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/utils/ResourceParser.java

```diff
@@ -17,7 +17,7 @@ public final class ResourceParser {
 
     // visible for testing
     public static void updateRoleNameAndInstance(
-        AbstractTelemetryBuilder builder, Resource resource, Configuration configuration) {
+        AbstractTelemetryBuilder builder, Resource resource, ConfigProperties configProperties) {
```

Comment:
Could it be package private (default visibility) with a test class having the same package?

Reply:
I think it needs to be public so we can call it from the Java agent code also.

also everything in Azure SDK `*.implementation.*` packages is considered implementation detail (non-public API), similar to `*.internal.*` used in OpenTelemetry